index.html

```diff
@@ -2866,32 +2836,57 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
         W3C TAG recommendations</a> to avoid using URNs in the Web.
         Therefore, Web NFC does not use the URN prefix neither when reading
-        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
-        before the type name.
+        or writing <a>NDEF messages</a>.
       </p>
       <p>
-        To <dfn>validate external type</dfn> given a |input:USVString| run
+        To <dfn>validate external type</dfn> given |input:USVString|, run
         these steps:
       </p>
       <ol class=algorithm id="validate-external-type">
         <li>
           If |input:string| is not a {{USVString}} or it is empty, or its
-          length exceeds 255 bytes, or it contains white spaces,
-          return `false`.
+          length exceeds 255 bytes, return `false`.
         </li>
         <li>
-          If |input| does not start with colon ('`:`'), return |input|.
-          <p class="issue">
-            Validating domain name needs an algorithm.
-          </p>
+          Find the last occurrence of `U+003A` (`:`) in |input|.
+          If not found, return `false`.
+        </li>
+        <li>
+          Let |domain| be the part of |input| preceding the last colon,
+          and |type| be the part following the last colon.
+        </li>
+        <li>
+          Run the following sub-steps of
+          <a href="https://url.spec.whatwg.org/#host-parsing">host parsing</a>
+          on |domain|:
+          <ul>
+            <li>
+              Let |asciiDomain| be the result of running
+              <a href="https://url.spec.whatwg.org/#concept-domain-to-ascii">
+              domain to ASCII</a> on |domain|.
+            </li>
+            <li>
+              If |asciiDomain| is failure, return `false`.
+            </li>
+            <li>
+              If |asciiDomain| contains a
+              <a href="https://url.spec.whatwg.org/#forbidden-host-code-point">
+              forbidden host code point</a>, return `false`.
+            </li>
+          </ul>
+        </li>
+        <li>
+          If |type| is empty, return `false`.
         </li>
         <li>
-          If |input| starts with colon ('`:`'), let |domain:string| be the
-          <a data-cite="url#concept-url-serializer">serialization</a> of the
-          <a data-cite="html">effective domain</a> of the {{Document/origin}}.
+          If |type| contains code points that are not [=ASCII alphanumeric=], or
+          `U+0024` (`$`), `U+0027` (`'`), `U+0028` (`(`), `U+0029` (`)`),
```

Comment:
interestingly URL specs writes out (() as LEFT PARENTESIS etc

Reply:
![image](https://user-images.githubusercontent.com/1101339/70896189-8c056f80-1ff0-11ea-95a6-2e81f6b3287f.png)

Reply:
We should probably follow for consistency

Reply:
Hmm. Strange. Should we use the same? (outsider reader might be confused why)

Reply:
You can say `U+0029 RIGHT PARENTHESIS ())` - that is allowed

Reply:
but I agree that it is ugly and that `U+0029 RIGHT PARENTHESIS` seems fine. We will link code points to URL spec anyway so we are good

Reply:
We definitely need to follow INFRA spec... everyone is moving to do that

Reply:
+1

Reply:
Fixed